Index: Code/BasicFilters/itkCropImageFilter.h
===================================================================
RCS file: /cvsroot/Insight/Insight/Code/BasicFilters/itkCropImageFilter.h,v
retrieving revision 1.6
diff -u -r1.6 itkCropImageFilter.h
--- Code/BasicFilters/itkCropImageFilter.h	10 Sep 2003 14:28:46 -0000	1.6
+++ Code/BasicFilters/itkCropImageFilter.h	24 Mar 2006 00:03:15 -0000
@@ -77,6 +77,32 @@
   itkGetMacro(UpperBoundaryCropSize, SizeType); 
   itkSetMacro(LowerBoundaryCropSize, SizeType);
   itkGetMacro(LowerBoundaryCropSize, SizeType);
+  void SetUpperBoundaryCropSize(const unsigned long bound)
+    {
+    SizeType s;
+    s.Fill( bound );
+    this->SetUpperBoundaryCropSize( s );
+    }
+    
+  void SetLowerBoundaryCropSize(const unsigned long bound)
+    {
+    SizeType s;
+    s.Fill( bound );
+    this->SetLowerBoundaryCropSize( s );
+    }
+    
+  void SetBoundaryCropSize(const SizeType & s)
+    {
+    this->SetUpperBoundaryCropSize( s );
+    this->SetLowerBoundaryCropSize( s );
+    }
+
+  void SetBoundaryCropSize(const unsigned long bound)
+    {
+    SizeType s;
+    s.Fill( bound );
+    this->SetBoundaryCropSize( s );
+    }
   
                  
 protected:
Index: Code/BasicFilters/itkPadImageFilter.h
===================================================================
RCS file: /cvsroot/Insight/Insight/Code/BasicFilters/itkPadImageFilter.h,v
retrieving revision 1.13
diff -u -r1.13 itkPadImageFilter.h
--- Code/BasicFilters/itkPadImageFilter.h	27 Apr 2005 12:54:57 -0000	1.13
+++ Code/BasicFilters/itkPadImageFilter.h	23 Mar 2006 07:44:06 -0000
@@ -18,6 +18,7 @@
 #define __itkPadImageFilter_h
 
 #include "itkImageToImageFilter.h"
+#include "itkSize.h"
 
 namespace itk
 {
@@ -78,6 +79,44 @@
   itkSetVectorMacro(PadUpperBound, const unsigned long, ImageDimension);
   itkGetVectorMacro(PadLowerBound, const unsigned long, ImageDimension);
   itkGetVectorMacro(PadUpperBound, const unsigned long, ImageDimension);
+  
+  void SetPadLowerBound(const itk::Size<ImageDimension> & bound)
+    {
+    this->SetPadLowerBound( bound.m_Size );
+    }
+    
+  void SetPadUpperBound(const itk::Size<ImageDimension> & bound)
+    {
+    this->SetPadUpperBound( bound.m_Size );
+    }
+    
+  void SetPadBound(const itk::Size<ImageDimension> & bound)
+    {
+    this->SetPadLowerBound( bound );
+    this->SetPadUpperBound( bound );    
+    }
+
+  void SetPadLowerBound(const unsigned long bound)
+    {
+    itk::Size<ImageDimension> s;
+    s.Fill( bound );
+    this->SetPadLowerBound( s );
+    }
+    
+  void SetPadUpperBound(const unsigned long bound)
+    {
+    itk::Size<ImageDimension> s;
+    s.Fill( bound );
+    this->SetPadUpperBound( s );
+    }
+    
+  void SetPadBound(const unsigned long bound)
+    {
+    itk::Size<ImageDimension> s;
+    s.Fill( bound );
+    this->SetPadBound( s );
+    }
+
                  
   /** PadImageFilter produces an image which is a different resolution
    * than its input image.  As such, PadImageFilter needs to
Index: Code/BasicFilters/itkRescaleIntensityImageFilter.txx
===================================================================
RCS file: /cvsroot/Insight/Insight/Code/BasicFilters/itkRescaleIntensityImageFilter.txx,v
retrieving revision 1.10
diff -u -r1.10 itkRescaleIntensityImageFilter.txx
--- Code/BasicFilters/itkRescaleIntensityImageFilter.txx	15 Jun 2005 16:13:48 -0000	1.10
+++ Code/BasicFilters/itkRescaleIntensityImageFilter.txx	23 Mar 2006 07:44:06 -0000
@@ -33,11 +33,11 @@
 RescaleIntensityImageFilter<TInputImage, TOutputImage>
 ::RescaleIntensityImageFilter()
 {
-  m_OutputMaximum   = NumericTraits<OutputPixelType>::Zero;
-  m_OutputMinimum   = NumericTraits<OutputPixelType>::max();
+  m_OutputMaximum   = NumericTraits<OutputPixelType>::max();
+  m_OutputMinimum   = NumericTraits<OutputPixelType>::NonpositiveMin();
 
-  m_InputMaximum   = NumericTraits<InputPixelType>::Zero;
-  m_InputMinimum   = NumericTraits<InputPixelType>::max();
+  m_InputMaximum   = NumericTraits<InputPixelType>::max();
+  m_InputMinimum   = NumericTraits<InputPixelType>::NonpositiveMin();
   
   m_Scale = 1.0;
   m_Shift = 0.0;
Index: Code/Common/itkIndex.h
===================================================================
RCS file: /cvsroot/Insight/Insight/Code/Common/itkIndex.h,v
retrieving revision 1.51
diff -u -r1.51 itkIndex.h
--- Code/Common/itkIndex.h	23 Aug 2005 01:25:37 -0000	1.51
+++ Code/Common/itkIndex.h	23 Mar 2006 07:44:08 -0000
@@ -239,7 +239,7 @@
    * \warning No bound checking is performed
    * \sa GetIndex()
    * \sa SetElement() */
-  IndexValueType GetElement( unsigned long element )
+  IndexValueType GetElement( unsigned long element ) const
     { return m_Index[ element ]; }
 
   /** Return a basis vector of the form [0, ..., 0, 1, 0, ... 0] where the "1"
Index: Code/Common/itkImageRegionConstIterator.h
===================================================================
RCS file: /cvsroot/Insight/Insight/Code/Common/itkImageRegionConstIterator.h,v
retrieving revision 1.20
diff -u -r1.20 itkImageRegionConstIterator.h
--- Code/Common/itkImageRegionConstIterator.h	31 Jan 2006 00:48:40 -0000	1.20
+++ Code/Common/itkImageRegionConstIterator.h	23 Mar 2006 07:44:07 -0000
@@ -281,6 +281,16 @@
     return *this;
   }
 
+  // rename some operators for wrapping
+  void increment()
+  {
+    this->operator++();
+  }
+  void decrement()
+  {
+    this->operator--();
+  }
+  
 private:
   void Increment(); // advance in a direction other than the fastest moving
   void Decrement(); // go back in a direction other than the fastest moving
Index: Code/Common/itkImageRegionConstIteratorWithIndex.h
===================================================================
RCS file: /cvsroot/Insight/Insight/Code/Common/itkImageRegionConstIteratorWithIndex.h,v
retrieving revision 1.10
diff -u -r1.10 itkImageRegionConstIteratorWithIndex.h
--- Code/Common/itkImageRegionConstIteratorWithIndex.h	16 Aug 2005 18:04:33 -0000	1.10
+++ Code/Common/itkImageRegionConstIteratorWithIndex.h	23 Mar 2006 07:44:07 -0000
@@ -186,6 +186,16 @@
    * will be set to be one pixel past the beginning of the region.
    * \sa operator++ */
   Self & operator--();
+
+  // rename some operators for wrapping
+  void increment()
+  {
+    this->operator++();
+  }
+  void decrement()
+  {
+    this->operator--();
+  }
 };
 
 /** \example itkImageIteratorTest.cpp */
Index: Code/Common/itkMatrix.h
===================================================================
RCS file: /cvsroot/Insight/Insight/Code/Common/itkMatrix.h,v
retrieving revision 1.20
diff -u -r1.20 itkMatrix.h
--- Code/Common/itkMatrix.h	18 Jul 2005 15:12:10 -0000	1.20
+++ Code/Common/itkMatrix.h	23 Mar 2006 07:44:08 -0000
@@ -160,6 +160,14 @@
   /** Copy constructor. */
   Matrix(const Self & matrix) : m_Matrix( matrix.m_Matrix ) {};
  
+ // Rename some operators for wrapping
+  inline T& Get( unsigned int row, unsigned int col )
+    { return m_Matrix(row,col); }
+
+  inline void Set( unsigned int row, unsigned int col, T value )
+    { m_Matrix(row,col) = value; }
+  
+  
 private:
   InternalMatrixType     m_Matrix;
 
