--- Code/Common/itkNumericTraitsVariableLengthVectorPixel.h.orig	2006-07-21 10:54:09.000000000 +0200
+++ Code/Common/itkNumericTraitsVariableLengthVectorPixel.h	2006-07-21 10:55:36.000000000 +0200
@@ -153,6 +153,7 @@
   typedef VariableLengthVector<unsigned long> AbsType;
   typedef VariableLengthVector<unsigned long> AccumulateType;
   typedef VariableLengthVector<double> RealType;
+  typedef double ScalarRealType;
   typedef VariableLengthVector<float> FloatType;
   static const VariableLengthVector<unsigned long> ITKCommon_EXPORT 
                         Zero( const VariableLengthVector< unsigned long > & );
