Index: itkCovariantVector.h
===================================================================
RCS file: /cvsroot/Insight/Insight/Code/Common/itkCovariantVector.h,v
retrieving revision 1.43
retrieving revision 1.44
diff -u -r1.43 -r1.44
--- itkCovariantVector.h	22 Jan 2006 17:29:45 -0000	1.43
+++ itkCovariantVector.h	21 Apr 2006 00:09:52 -0000	1.44
@@ -3,8 +3,8 @@
   Program:   Insight Segmentation & Registration Toolkit
   Module:    $RCSfile: itkCovariantVector.h,v $
   Language:  C++
-  Date:      $Date: 2006/01/22 17:29:45 $
-  Version:   $Revision: 1.43 $
+  Date:      $Date: 2006/04/21 00:09:52 $
+  Version:   $Revision: 1.44 $
 
   Copyright (c) Insight Software Consortium. All rights reserved.
   See ITKCopyright.txt or http://www.itk.org/HTML/Copyright.htm for details.
@@ -247,6 +247,7 @@
 
 } // end namespace itk
   
+#include "itkNumericTraitsCovariantVectorPixel.h"
 
 #ifndef ITK_MANUAL_INSTANTIATION
 #include "itkCovariantVector.txx"
Index: itkRGBPixel.h
===================================================================
RCS file: /cvsroot/Insight/Insight/Code/Common/itkRGBPixel.h,v
retrieving revision 1.36
retrieving revision 1.37
diff -u -r1.36 -r1.37
--- itkRGBPixel.h	12 Aug 2005 16:16:03 -0000	1.36
+++ itkRGBPixel.h	21 Apr 2006 00:09:54 -0000	1.37
@@ -3,8 +3,8 @@
   Program:   Insight Segmentation & Registration Toolkit
   Module:    $RCSfile: itkRGBPixel.h,v $
   Language:  C++
-  Date:      $Date: 2005/08/12 16:16:03 $
-  Version:   $Revision: 1.36 $
+  Date:      $Date: 2006/04/21 00:09:54 $
+  Version:   $Revision: 1.37 $
 
   Copyright (c) Insight Software Consortium. All rights reserved.
   See ITKCopyright.txt or http://www.itk.org/HTML/Copyright.htm for details.
@@ -150,6 +150,8 @@
 
 } // end namespace itk
 
+#include "itkNumericTraitsRGBPixel.h"
+
 #ifndef ITK_MANUAL_INSTANTIATION
 #include "itkRGBPixel.txx"
 #endif
Index: itkSymmetricSecondRankTensor.h
===================================================================
RCS file: /cvsroot/Insight/Insight/Code/Common/itkSymmetricSecondRankTensor.h,v
retrieving revision 1.21
retrieving revision 1.22
diff -u -r1.21 -r1.22
--- itkSymmetricSecondRankTensor.h	29 Jun 2005 20:43:38 -0000	1.21
+++ itkSymmetricSecondRankTensor.h	21 Apr 2006 00:09:54 -0000	1.22
@@ -3,8 +3,8 @@
   Program:   Insight Segmentation & Registration Toolkit
   Module:    $RCSfile: itkSymmetricSecondRankTensor.h,v $
   Language:  C++
-  Date:      $Date: 2005/06/29 20:43:38 $
-  Version:   $Revision: 1.21 $
+  Date:      $Date: 2006/04/21 00:09:54 $
+  Version:   $Revision: 1.22 $
 
   Copyright (c) Insight Software Consortium. All rights reserved.
   See ITKCopyright.txt or http://www.itk.org/HTML/Copyright.htm for details.
@@ -192,6 +192,8 @@
 
 } // end namespace itk
 
+#include "itkNumericTraitsTensorPixel.h"
+
 #ifndef ITK_MANUAL_INSTANTIATION
 #include "itkSymmetricSecondRankTensor.txx"
 #endif
Index: itkVariableLengthVector.h
===================================================================
RCS file: /cvsroot/Insight/Insight/Code/Common/itkVariableLengthVector.h,v
retrieving revision 1.7
retrieving revision 1.8
diff -u -r1.7 -r1.8
--- itkVariableLengthVector.h	10 Mar 2006 18:13:40 -0000	1.7
+++ itkVariableLengthVector.h	21 Apr 2006 00:09:54 -0000	1.8
@@ -3,8 +3,8 @@
   Program:   Insight Segmentation & Registration Toolkit
   Module:    $RCSfile: itkVariableLengthVector.h,v $
   Language:  C++
-  Date:      $Date: 2006/03/10 18:13:40 $
-  Version:   $Revision: 1.7 $
+  Date:      $Date: 2006/04/21 00:09:54 $
+  Version:   $Revision: 1.8 $
 
   Copyright (c) Insight Software Consortium. All rights reserved.
   See ITKCopyright.txt or http://www.itk.org/HTML/Copyright.htm for details.
@@ -425,6 +425,7 @@
 
 } // namespace itk
 
+#include "itkNumericTraitsVariableLengthVectorPixel.h"
 
 #ifndef ITK_MANUAL_INSTANTIATION
 #include "itkVariableLengthVector.txx"
Index: itkVector.h
===================================================================
RCS file: /cvsroot/Insight/Insight/Code/Common/itkVector.h,v
retrieving revision 1.76
retrieving revision 1.77
diff -u -r1.76 -r1.77
--- itkVector.h	20 Apr 2006 14:54:10 -0000	1.76
+++ itkVector.h	21 Apr 2006 00:09:54 -0000	1.77
@@ -3,8 +3,8 @@
   Program:   Insight Segmentation & Registration Toolkit
   Module:    $RCSfile: itkVector.h,v $
   Language:  C++
-  Date:      $Date: 2006/04/20 14:54:10 $
-  Version:   $Revision: 1.76 $
+  Date:      $Date: 2006/04/21 00:09:54 $
+  Version:   $Revision: 1.77 $
 
   Copyright (c) Insight Software Consortium. All rights reserved.
   See ITKCopyright.txt or http://www.itk.org/HTML/Copyright.htm for details.
@@ -254,6 +254,9 @@
 
 } // end namespace itk
 
+#include "itkNumericTraitsVectorPixel.h"
+
+
 // Define instantiation macro for this template.
 #define ITK_TEMPLATE_Vector(_, EXPORT, x, y) namespace itk { \
   _(2(class EXPORT Vector< ITK_TEMPLATE_2 x >)) \
